openedx/core/djangoapps/content_tagging/tests/test_api.py

```diff
@@ -248,3 +253,200 @@ def test_cannot_tag_across_orgs(self):
+                self.taxonomy_2.id: list(sequential_tags2),
+            },
+        )
+
+        assert self.expected_tagged_xblock.children is not None  # type guard
+        self.expected_tagged_xblock.children.append(tagged_sequential)
+
+        vertical = self.store.create_child(self.user_id, self.sequential.location, "vertical", "test_vertical1")
+        vertical_tags = api.tag_content_object(
+            object_key=vertical.location,
+            taxonomy=self.taxonomy_2,
+            tags=['Tag 2.2'],
+        )
+        tagged_vertical = TaggedContent(
+            xblock=self.store.get_item(vertical.location),
+            children=[],
+            object_tags={
+                self.taxonomy_2.id: list(vertical_tags),
+            },
+        )
+
+        assert tagged_sequential.children is not None  # type guard
+        tagged_sequential.children.append(tagged_vertical)
+
+        vertical2 = self.store.create_child(self.user_id, self.sequential.location, "vertical", "test_vertical2")
+        tagged_vertical2 = TaggedContent(
+            xblock=self.store.get_item(vertical2.location),
+            children=[],
+            object_tags={},
+        )
+        assert tagged_sequential.children is not None  # type guard
+        tagged_sequential.children.append(tagged_vertical2)
+
+        text = self.store.create_child(self.user_id, vertical2.location, "html", "test_html")
+        text_tags = api.tag_content_object(
+            object_key=text.location,
+            taxonomy=self.taxonomy_2,
+            tags=['Tag 2.1'],
+        )
+        tagged_text = TaggedContent(
+            xblock=self.store.get_item(text.location),
+            children=[],
+            object_tags={
+                self.taxonomy_2.id: list(text_tags),
+            },
+        )
+
+        assert tagged_vertical2.children is not None  # type guard
+        tagged_vertical2.children.append(tagged_text)
+
+
+@ddt.ddt
+class TestContentTagChildrenExport(TaggedCourseMixin):  # type: ignore[misc]
+    """
+    Test exporting content objects
+    """
+    def _compare_tagged_xblock(self, expected: TaggedContent, actual: TaggedContent):
+        """
+        Compare two TaggedContent objects
+        """
+        assert expected.xblock.location == actual.xblock.location
+        assert expected.object_tags == actual.object_tags
+        if expected.children is None:
+            assert actual.children is None
+            return
+
+        assert actual.children is not None
+        for i in range(len(expected.children)):
+            self._compare_tagged_xblock(expected.children[i], actual.children[i])
+
+    @ddt.data(
+        True,
+        False,
+    )
+    def test_export_tagged_course(self, include_children: bool) -> None:
+        """
+        Test if we can export a course
+        """
+        tagged_xblock, taxonomies = api.get_content_tags_for_object(self.course.id, include_children=include_children)
```

Comment:
Could you also add some tests that check the number of queries required to use this api?

Reply:
Done: 484c0424afb23d785cec0941f7d534379fb15bb2!